shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -670,15 +673,15 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
       sym.body.collectTypeNames.flatMap(resolveTraitFields)
 
     val ctorParams = fields.map { f =>
-      memberList += nuTypeScope.declareNewClassMember(f, Some(false), false)
+      memberList += NewClassMemberSymbol(f, Some(false), false).tap(nuTypeScope.register(_))
```

Comment:
I think it may work like this:
```suggestion
      memberList += NewClassMemberSymbol(f, Some(false), false).tap(nuTypeScope.register)
```